src/modules/85-cv/components/HarnessServiceAndEnvironment/components/ServiceSelectOrCreate/ServiceSelectOrCreate.tsx

```diff
@@ -69,27 +79,46 @@ export const ServiceSelectOrCreate: React.FC<ServiceSelectOrCreateProps> = props
     customLoading
   })
 
-  const onSelectChange = (val: SelectOption): void => {
-    if (val.value === ADD_NEW_VALUE) {
+  const onSelectChange = (val: SelectOption | string): void => {
+    if (typeof val !== 'string' && val?.value === ADD_NEW_VALUE) {
       openHarnessServiceModal()
     } else {
-      props.onSelect(val)
+      if (typeof val === 'string') {
+        props.onSelect({ label: val, value: val })
+      } else {
+        props.onSelect(val)
+      }
     }
   }
 
   return (
     <Container onClick={e => e.stopPropagation()}>
-      <Select
-        name={name ?? 'service'}
-        value={props.item}
-        className={props.className}
-        disabled={props.disabled}
-        items={selectOptions}
-        inputProps={{
-          placeholder: loading ? getString('loading') : placeholder ?? getString('cv.selectCreateService')
-        }}
-        onChange={onSelectChange}
-      />
+      {isMultiType ? (
+        <MultiTypeInput
+          name={name ?? 'service'}
+          disabled={props.disabled}
+          selectProps={{
+            items: selectOptions
+          }}
+          allowableTypes={[MultiTypeInputType.FIXED, MultiTypeInputType.RUNTIME]}
+          value={props.item}
+          style={{ width: '400px' }}
+          onChange={(val: any) => onSelectChange(val)}
```

Comment:
Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9235241)
```suggestion
          onChange={(val: unknown) => onSelectChange(val)}
```